### ISSUE TITLE
Mark `t.i.process.Process` and `t.i.process.PTYProcess` as implementing IProcessTransport

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4585]
|<img alt="Screwtape's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Screwtape reported|
|-|-|
|Trac ID|trac#4585|
|Type|enhancement|
|Created|2010-07-14 01:30:28Z|
|Branch|https://github.com/twisted/twisted/tree/process-interface-4585|

Currently the [IProcessTransport documentation](http://twistedmatrix.com/documents/current/api/twisted.internet.interfaces.IProcessTransport.html) claims that the only process transport in Twisted is for Win32, which is a total lie. `t.i.process.Process` and `t.i.process.PTYProcess` also implement it, and are very likely much more solid implementations.

We should mark both those classes as implementing that interface.

(this was filed once before as [#4479](https://github.com/twisted/twisted/issues/4479), but that ticket got derailed into a different documentation issue)

<details><summary>Searchable metadata</summary>

```
trac-id__4585 4585
type__enhancement enhancement
reporter__Screwtape Screwtape
priority__normal normal
milestone__ 
branch__branches_process_interface_4585 branches/process-interface-4585
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1279071028000000 1279071028000000
changetime__1286801160000000 1286801160000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

[#4648](https://github.com/twisted/twisted/issues/4648) was a duplicate of this.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [30045]) Branching to 'process-interface-4585'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [30046]) Add a test verifying the process transport provides IProcessTransport

refs [#4585](https://github.com/twisted/twisted/issues/4585)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [30047]) Make Process and PTYProcess implement IProcessTransport

refs [#4585](https://github.com/twisted/twisted/issues/4585)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/process-interface-4585)

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_6"></a>|@mwhudson set owner to @exarkun|
|-|-|

test_process imports verifyObject and then doesn't use it.

I think "# Oh, the transport is available synchronously, how convenient." is a bit, um, sort of passive-aggressive or something.  Can you make it a bit less snarky?

Fix these and land, please.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [30127]) Remove unused import; adjust snarky comment

refs [#4585](https://github.com/twisted/twisted/issues/4585)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [30128]) Merge process-interface-4585

Author: exarkun
Reviewer: mwhudson
Fixes: [#4585](https://github.com/twisted/twisted/issues/4585)

Declare that the POSIX process transports (for PTYs and non-PTYs) implement
`IProcessTransport`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
